src/baldr/graphtile.cc

```diff
@@ -434,6 +437,9 @@ GraphId GraphTile::GetTileId(const std::string& fname) {
                                                       '7',
                                                       '8',
                                                       '9'};
+  auto fname = fname_in;
+   // now update path separator based on OS..
+  std::replace(fname.begin(), fname.end(), '/', filesystem::path::preferred_separator);                                                    
```

Comment:
we should probably fix this upstream too right? i had thought most instances of calls to this were with paths from the filesystem so i expected them to have the right separator. i am guess that perhaps the fuction you patched above might get sent here as well? i need ot have a closer look

Reply:
ok i see this is coming from the tar file itself since im guessing tars dont automatically switch the separator they use. i think the best fix would be to have graph reader which looks at paths from our tar parser do the switch when we set the keys in the unordered map.

so what happens is first we parse the tar and read out all the file entries for all the contents of the tar: https://github.com/valhalla/valhalla/blob/master/valhalla/midgard/sequence.h#L609

that line there takes the tar file path (likely using `/` regardless of platform) and stores it as the key we use to talk about files in the tar. then over here: https://github.com/valhalla/valhalla/blob/master/src/baldr/graphreader.cc#L34 we keep a map of the tile id to the pointer in the memory map.

i think right either here in graphreader or in the sequence you can do the replacement instead of inside `GetTileId` and put a note that tars apparently dont use OS specific separators for their entries. that way the fix is closer to the source of the problem

Reply:
Yeah that's probably a better place. Will add in a couple lines in sequence.h and revert that change to graphtile.